src/main/java/iudx/resource/server/metering/util/QueryBuilder.java

```diff
@@ -87,37 +91,105 @@ public JsonObject buildReadingQuery(JsonObject request) {
 
     long toTime = getEpochTime(endZDT);
 
-    if (request.getString(HEADER_OPTIONS) != null) {
-      if (providerID != null) {
+      if (providerID != null)
         query =
             new StringBuilder(
-                PROVIDERID_TIME_INTERVAL_COUNT_QUERY
+                PROVIDERID_TIME_INTERVAL_READ_QUERY
                     .replace("$0", databaseTableName)
                     .replace("$1", Long.toString(fromTime))
                     .replace("$2", Long.toString(toTime))
                     .replace("$3", providerID));
-      } else {
+      else
```

Comment:
missing `{`  `}` braces, better to write if else inside braces.

Reply:
done